sphinx/util/requests.py

```diff
@@ -50,6 +62,25 @@ def head(url: str, **kwargs: Any) -> requests.Response:
 
 
 class _Session(requests.Session):
+    def get_redirect_target(self, resp: requests.Response) -> str | None:
+        """Overrides the default requests.Session.get_redirect_target"""
+        # do not follow redirections that match ignored URI patterns
+        if resp.is_redirect:
+            location = resp.headers['location']
+            destination = urljoin(resp.url, location)
```

Comment:
`location` is a single-use variable with no side-effects, so it should probably be factored-out.